lib/shared_preferences.dart

```diff
@@ -0,0 +1,32 @@
+
+import 'package:flutter_pk/global.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SharedPreferencesHandler {
+  static Future setPreference(String key, String value) async {
+    final sharedPrefsFuture = sharedPreferences;
+    final waitFuture = Future.delayed(Duration(seconds: 5));
```

Comment:
This look meaningless here. It's a UI thing to wait for 5 seconds but looks out of context here. I see first 3-4 lines repeating in all methods. We also don't need to use static methods. Create a variable at class level that could hold SharedPreferences object and use it in all the methods. We can use lazy loading to initialize that variable.